diversity-lunch-fe/src/data/generic/GenericFetch.ts

```diff
@@ -8,109 +7,128 @@ import {
+                    try {
+                        const result: T = await response.json();
+                        dispatch(this.add([result]));
+                    } catch {
+                        callbacks.onNetworkError(new Error("Couldn't parse response body to json."));
+                    }
+                } else if (callbacks.statusCodeHandlers[statusCode]) {
+                    callbacks.statusCodeHandlers[statusCode]!(response);
                 }
             } catch (error) {
-                dispatch(this._errorSlice.error(undefined));
+                callbacks.onNetworkError(error as Error);
             }
         };
     }
 
-    public put(item : T) {
+    public put(item : T, callbacks: FetchCallbacks) {
         return async (dispatch: Dispatch) => {
             try {
                 const response = await authenticatedFetchPut(this.url + this.endpoint, item);
+                const statusCode: StatusCode = response.status.toString() as StatusCode;
 
-                if (!response.ok) {
-                    dispatch(this._errorSlice.httpError({ statusCode: response.status }));
-                } else {
-                    const result : T = await response.json();
-                    dispatch(this.update([result]));
+                if (response.ok) {
+                    try {
+                        const result: T = await response.json();
+                        dispatch(this.update([result]));
+                    } catch {
+                        callbacks.onNetworkError(new Error("Couldn't parse response body to json."));
+                    }
+                } else if (callbacks.statusCodeHandlers[statusCode]) {
+                    callbacks.statusCodeHandlers[statusCode]!(response);
                 }
             } catch (error) {
-                dispatch(this._errorSlice.error(undefined));
+                callbacks.onNetworkError(error as Error);
             }
         };
     }
 
-    public removeById(id : number) {
+    public removeById(id : number, callbacks: FetchCallbacks) {
         return async (dispatch: Dispatch) => {
             try {
                 const response = await authenticatedFetchDelete(this.url + this.endpoint + id);
+                const statusCode: StatusCode = response.status.toString() as StatusCode;
 
-                if (!response.ok) {
-                    dispatch(this._errorSlice.httpError({ statusCode: response.status }));
-                } else {
-                    // const result : T[] = await response.json();
+                if (response.ok) {
                     dispatch(this.remove([id]));
+                } else if (callbacks.statusCodeHandlers[statusCode]) {
+                    callbacks.statusCodeHandlers[statusCode]!(response);
                 }
             } catch (error) {
-                dispatch(this._errorSlice.error(undefined));
+                callbacks.onNetworkError(error as Error);
             }
         };
     }
```

Comment:
Kann der Rahmen in eine Methode ausgelagert werden? Hier ist irgendwie viel Copy-Paste Code (Bezieht sich auf die ganzen Funktionen)